ast/check.go

```diff
@@ -36,7 +37,7 @@ func newTypeChecker() *typeChecker {
 // CheckBody runs type checking on the body and returns a TypeEnv if no errors
 // are found. The resulting TypeEnv wraps the provided one. The resulting
 // TypeEnv will be able to resolve types of vars contained in the body.
-func (tc *typeChecker) CheckBody(env *TypeEnv, body Body) (*TypeEnv, Errors) {
+func (tc *typeChecker) CheckBody(env *TypeEnv, body Body, funcs map[String]*Func) (*TypeEnv, Errors) {
```

Comment:
So the reason for passing that is that I didn't globally register functions as builtins (as per I conversation I remember us having), and instead had a compiler specific map that was passed around. But if there's no problem with globally registering them, I think that would clean up the code a lot.